libethcore/SealEngine.cpp

```diff
@@ -55,6 +56,19 @@ void SealEngineFace::verifyTransaction(ImportRequirements::value _ir, Transactio
 
 	if (_header.number() >= chainParams().homesteadForkBlock && (_ir & ImportRequirements::TransactionSignatures) && _t.hasSignature())
 		_t.checkLowS();
+
+    eth::EVMSchedule const& schedule = evmSchedule(_header.number());
+
+    // Pre calculate the gas needed for execution
+    if (_t.baseGasRequired(schedule) > _t.gas())
```

Comment:
This was earlier checked only if `_ir & ImportRequirements::TransactionBasic` - basic checks requested. Any reason why did you drop this condition?

Reply:
hm. it looks like there is not. should return it back.